### PR TITLE
Add activity type and workflow type tags for activity e2e metrics

### PR DESCRIPTION
**Pull request by meiliang86**

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java [hunk 1]
```diff
@@ -155,43 +150,52 @@ private TaskHandlerImpl(ActivityTaskHandler handler) {
...
+    public void handle(PollForActivityTaskResponse task) throws Exception {
       Scope metricsScope =
           options
               .getMetricsScope()
               .tagged(
-                  ImmutableMap.of(MetricsTag.ACTIVITY_TYPE, task.task.getActivityType().getName()));
+                  ImmutableMap.of(
+                      MetricsTag.ACTIVITY_TYPE,
+                      task.getActivityType().getName(),
+                      MetricsTag.WORKER_TYPE,
```
Comment (vancexu) [inline src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java]: Should this be `WorkflowType` or `WORKER_TYPE`?

File: src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java [hunk 2]
```diff
@@ -155,43 +150,52 @@ private TaskHandlerImpl(ActivityTaskHandler handler) {
...
         sw.stop();
 
         sw = metricsScope.timer(MetricsType.ACTIVITY_RESP_LATENCY).start();
-        sendReply(task.task, response, metricsScope);
+        sendReply(task, response, metricsScope);
         sw.stop();
 
-        task.markDone();
+        metricsScope
+            .timer(MetricsType.ACTIVITY_E2E_LATENCY)
```
Comment (vancexu) [inline src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java]: Looks like `ACTIVITY_E2E_LATENCY` is already recoded in activity poll method. 
The other concern on `System.nanoTime() - task.getScheduledTimestampOfThisAttempt())` might not be accurate considering time skew between server and worker

Comment (meiliang86) [reply src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java]: It's moved from there. Reason is that this is more accurate (e2e time should start from schedule time). Yes I know about time skew, but that's the best we can do.

File: src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java (hunk 1 above)
Comment (meiliang86) [reply src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java]: Yes it should be workflow type. Fixed.

File: src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java [hunk 3]
```diff
@@ -155,43 +150,52 @@ private TaskHandlerImpl(ActivityTaskHandler handler) {
...
-      propagateContext(task.task);
+      propagateContext(task);
 
       try {
         Stopwatch sw = metricsScope.timer(MetricsType.ACTIVITY_EXEC_LATENCY).start();
-        ActivityTaskHandler.Result response = handler.handle(task.task, metricsScope, false);
+        ActivityTaskHandler.Result response = handler.handle(task, metricsScope, false);
         sw.stop();
 
         sw = metricsScope.timer(MetricsType.ACTIVITY_RESP_LATENCY).start();
```
Comment (dhiren-singh) [inline src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java]: Manual activity completion isn't captured by these metrics.

File: src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java [hunk 4]
```diff
@@ -155,43 +150,52 @@ private TaskHandlerImpl(ActivityTaskHandler handler) {
...
+      MDC.put(LoggerTag.ACTIVITY_ID, task.getActivityId());
+      MDC.put(LoggerTag.ACTIVITY_TYPE, task.getActivityType().getName());
+      MDC.put(LoggerTag.WORKFLOW_ID, task.getWorkflowExecution().getWorkflowId());
+      MDC.put(LoggerTag.RUN_ID, task.getWorkflowExecution().getRunId());
 
-      propagateContext(task.task);
+      propagateContext(task);
 
       try {
         Stopwatch sw = metricsScope.timer(MetricsType.ACTIVITY_EXEC_LATENCY).start();
```
Comment (dhiren-singh) [inline src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java]: Manual activity completion isn't captured by these metrics.

**Pull request merged**